templates/etc/systemd/system/etcd.service.j2

```diff
@@ -4,9 +4,10 @@
 {{hostvars[host]['ansible_hostname']}}=https://{{hostvars[host]['ansible_' + etcd_interface].ipv4.address}}:{{etcd_peer_port}}{% if not loop.last %},{% endif %}
 {%- endfor -%} 
 {%- endmacro -%}
-
+{%- if not etcd_settings['discovery-srv'] %}
```

Comment:
`initial-cluster` must be removed (`__delitem__`) from `etcd_settings` when `discovery-srv` is configured?

Reply:
Yes, they're incompatible options

Reply:
To expand on this, there are three exclusive bootstrapping mechanisms for etcd:
- static with `initial-cluster` definition
- etcd discovery (requires an already working external etcd cluster, not in scope)
- DNS discovery with `discovery-srv`

Ref: https://etcd.io/docs/v3.5/op-guide/clustering/

Reply:
`initial-cluster` should not be set manually but computed by the macro above from ansible inventory, the key doesn't even exist in defaults, that's why I didn't add an `else` statement. For that same reason I can't `__delitem__` it, but I can `pop` it to ensure exclusiveness.